install/installing-custom-components.md

```diff
@@ -0,0 +1,198 @@
+[automatic sidecar injection](https://istio.io/docs/setup/kubernetes/sidecar-injection).
+You should install the default package if you are just getting started with Knative.
+
+Alternatively, you can choose to install Istio with manual sidecar injection or with support for
+[Custom Resource Definitions (CRD)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
+
+### Choosing a Istio installation package
+
+You can choose from the following installation packages to install the Istio
+service mesh on you cluster:
+
+| Istio Package Filename | Description       |
+| ---------------------- | ----------------- |
+| [`istio.yaml`][a]             | Default: Configured for Automatic Sidecar Injection. |
+| [`istio-crds.yaml`][b]        | Support for Custom Resource Definitions. **Q: This have to do with "Pilot"? Why choose this?** |
+| [`istio-lean.yaml`][c]        | Configured for Manual Sidecar Injection. **Q: When or why should users choose this?** |
+
+[a]: https://github.com/knative/serving/releases/download/v0.2.1/istio.yaml
+[b]: https://github.com/knative/serving/releases/download/v0.2.1/istio-crds.yaml
+[c]: https://github.com/knative/serving/releases/download/v0.2.1/istio-lean.yaml
+
+### Installing Istio installation packages
+
+1. Install a Istio package with the `kubectl apply` command by specifying the
+   package filename:
+    ```bash
+    kubectl apply --filename https://github.com/knative/serving/releases/download/v0.2.1/[FILENAME].yaml
+    ```
+
+    where `[FILENAME]` is the name of the Istio package that you want to install. Example:
+    `https://github.com/knative/serving/releases/download/v0.2.1/istio.yaml
+    `
+1. Label the default namespace with `istio-injection=enabled`: **Q: For which install package is this needed? Or is this needed for all Istio installs?**
+    ```bash
+    kubectl label namespace default istio-injection=enabled
+    ```
+1. View the status of your Istio installation. It might take a few
+   seconds so rerun the following command until all of the pods show a
+   `STATUS` of `Running` or `Completed`:
+    ```bash
+    kubectl get pods --namespace istio-system
+    ```
+
+    > Tip: You can append the `--watch` flag to the `kubectl get` commands to
+      view the pod status in realtime. You use `CTRL + C` to exit watch mode.
+
+## Installing Knative components
+
+You can choose to separately install only the individual Knative components
+that you want. You can also later add a pre-configured observability plugin to enable
+monitoring, logging, and tracing in your cluster.
+
+Installing individual Knative components excludes other features like
+observabitliy plugins. Therefore, you must separately install and run the
+`kubectl apply` command for each component or plugin that you want installed.
+
+### Choosing Knative installation packages
+
+The following section describes all the Knative installation packages from which
+you can choose.
```

Comment:
"The following section describes all the Knative installation packages from which
you can choose." --> "The following Knative installation packages are available."

Reply:
done